### PR TITLE
style(ui): truncate project name and description

### LINKED ISSUE #1075: Fix layouts to acommodate long user and project names
Follow up of: https://github.com/radicle-dev/radicle-upstream/issues/954.

We have the following issues with responsiveness and long titles:

Screenshot 2020-10-19 at 11 48 34
Screenshot 2020-10-19 at 11 49 30
Screenshot 2020-10-19 at 11 49 40

Screenshot 2020-10-19 at 11 48 25

### PR DESCRIPTION
**Pull request by juliendonck**
Project name and description take as much space as they can but stay on 1 line. Multi line truncation is much harder and not sure if it's worth it before β1...
closes #1075

### CODE REVIEW

**Comment by juliendonck** [comment]
@rudolfs this fixes  the names and descriptions but to truncate the URN we need to get a lot smarter. Not sure if it's worth it.

**Review by MeBrei — APPROVED** [review]

**Comment by rudolfs** [comment]
When the window is resized (~900px) and we scroll around the layout looks broken:
Screenshot 2020-10-27 at 08 46 22
Screenshot 2020-10-27 at 08 46 04

Ellipsis is missing in project description in the project listing:
Screenshot 2020-10-27 at 08 44 03

Might be nice to add `title` attributes to the wrapped project name and description, otherwise there's no way to see the full one if they're long:
```diff
--- a/ui/DesignSystem/Component/Header/Large.svelte
+++ b/ui/DesignSystem/Component/Header/Large.svelte
@@ -111,10 +111,10 @@
       {/if}

       <div class="metadata">
-        <h1 data-cy="entity-name" class="typo-overflow-ellipsis">{name}</h1>
+        <h1 data-cy="entity-name" class="typo-overflow-ellipsis" title={name}>{name}</h1>
         <ShareableIdentifier {urn} style="margin-top: 0.5rem;" />
         {#if description.length > 0}
-          <p class="description typo-overflow-ellipsis">{description}</p>
+          <p class="description typo-overflow-ellipsis" title={description}>{description}</p>
         {/if}
         {#if stats}
           <div class="project-stats" data-cy="project-stats">
```

Screenshot 2020-10-27 at 08 49 37
Screenshot 2020-10-27 at 08 49 43

**Comment by juliendonck** [comment]
@rudolfs made it so much smarter, you'd be proud.

**Comment by juliendonck** [comment]
Screenshot 2020-10-27 at 14 37 06

just saying

File: ui/Screen/NotFound.svelte [hunk 1]
```diff
@@ -1,10 +1,9 @@
 <script>
-  import { Flex } from "../DesignSystem/Primitive";
   import { SidebarLayout } from "../DesignSystem/Component";
 </script>
 
 <SidebarLayout>
-  <Flex align="left">
+  <div class="display: flex; flex: 1;">
```
Comment (rudolfs) [inline ui/Screen/NotFound.svelte:6]: 👀

File: ui/DesignSystem/Primitive/index.js [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import Avatar from "./Avatar.svelte";
 import Button from "./Button.svelte";
 import Emoji from "./Emoji.svelte";
-import Flex from "./Flex.svelte";
```
Comment (rudolfs) [inline ui/DesignSystem/Primitive/index.js:4]: Don't forget to delete the file itself.

File: ui/DesignSystem/Component/Header/Large.svelte [hunk 3]
```diff
@@ -117,27 +112,27 @@
...
+          <p class="description typo-overflow-ellipsis" title={description}>
+            {description}
+          </p>
         {/if}
         {#if stats}
           <div class="project-stats" data-cy="project-stats">
             <div class="project-stat-item">
               <Icon.Branch />
-              <p style="margin-left: 0.5rem;">
+              <p style="margin-left: 0.5rem; white-space: nowrap;">
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Header/Large.svelte:128]: Can we move the styles for `p` into `<style>`?

File: ui/DesignSystem/Component/Header/Large.svelte [hunk 4]
```diff
@@ -117,27 +112,27 @@
...
                 {stats.branches === 1 ? `1 Branch` : `${stats.branches} Branches`}
               </p>
             </div>
             <span class="typo-mono-bold project-stat-separator">•</span>
             <div class="project-stat-item">
               <Icon.User />
-              <p style="margin-left: 0.5rem;">
+              <p style="margin-left: 0.5rem; white-space: nowrap;">
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Header/Large.svelte:135]: Ditto.

File: ui/DesignSystem/Component/ProjectCard.svelte [hunk 5]
```diff
@@ -12,21 +12,17 @@
...
     max-height: 3.2rem;
-  }
-
-  .container {
-    display: flex;
-    flex: 1;
     flex-direction: column;
     justify-content: center;
-    width: 10%;
+    min-width: 0;
```
Comment (rudolfs) [inline ui/DesignSystem/Component/ProjectCard.svelte:18]:
```suggestion
```

File: ui/Screen/Project/Commit.svelte [hunk 6]
```diff
@@ -52,6 +52,13 @@
     color: var(--color-foreground-level-6);
   }
 
+  .context {
+    align-items: flex-end;
+
```
Comment (rudolfs) [inline ui/Screen/Project/Commit.svelte:57]:
```suggestion
```

File: ui/DesignSystem/Component/Header/Large.svelte [hunk 7]
```diff
@@ -36,27 +36,22 @@
...
+    min-width: 16rem;
+    margin-right: 1.5rem;
   }
 
   .metadata {
     display: flex;
     flex-direction: column;
     align-self: center;
+    width: -webkit-fill-available;
+    min-width: 0;
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Header/Large.svelte:48]:
```suggestion
```

File: ui/DesignSystem/Component/ProjectList.svelte [hunk 8]
```diff
@@ -17,25 +16,32 @@
...
 </script>
 
+<style>
+  .list-item {
+    display: flex;
+    width: 100%;
+    justify-content: space-between;
+    padding: 1.375rem 1.5rem;
+    align-items: center;
+    min-width: 0;
```
Comment (rudolfs) [inline ui/DesignSystem/Component/ProjectList.svelte:26]:
```suggestion
```

**Review by rudolfs — APPROVED** [review]
Looks good.

Added some nits.

**Review by rudolfs — DISMISSED** [review]

File: ui/DesignSystem/Component/Header/Large.svelte (hunk 4 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/Header/Large.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1106/commits/2e74eea5fe4d687219cf9314f50d50ef622be4a0

File: ui/DesignSystem/Component/Header/Large.svelte (hunk 3 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/Header/Large.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1106/commits/2e74eea5fe4d687219cf9314f50d50ef622be4a0

**Review by rudolfs — APPROVED** [review]

**Review by MeBrei — APPROVED** [review]
⭐

**Pull request merged**